src/Table/index.jsx

```diff
@@ -1,16 +1,110 @@
 import React from 'react';
 import classNames from 'classnames';
+import FontAwesomeStyles from 'font-awesome/css/font-awesome.min.css';
+import isRequiredIf from 'react-proptype-conditional-require';
 import PropTypes from 'prop-types';
 
 import styles from './Table.scss';
+import Button from '../Button';
 
 class Table extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = props.tableSortable ?
+      {
+        sortedColumn: this.props.defaultSortedColumn,
+        sortDirection: this.props.defaultSortDirection,
+      }
+      :
+      {
+        sortedColumn: '',
+        sortDirection: '',
+      };
+
+    this.onSortClick = this.onSortClick.bind(this);
+  }
+
+  onSortClick(columnKey) {
+    let newDirection = 'desc';
+
+    if (this.state.sortedColumn === columnKey) {
+      newDirection = (this.state.sortDirection === 'desc' ? 'asc' : 'desc');
+
+      this.setState({
+        sortDirection: newDirection,
+      });
+    } else {
+      this.setState({
+        sortedColumn: columnKey,
+        sortDirection: newDirection,
+      });
+    }
+
+    const currentlySortedColumn = this.props.columns.find(
+      column => (columnKey === column.key));
+    currentlySortedColumn.onSort(newDirection);
+  }
+
   getCaption() {
     return this.props.caption && (
       <caption>{this.props.caption}</caption>
     );
   }
 
+  getHeadingText(columnKey) {
+    let text = '';
+
+    if (this.state.sortedColumn === columnKey) {
+      text = this.state.sortDirection === 'desc' ? 'sort descending' : 'sort ascending';
+    } else {
+      text = 'click to sort';
+    }
+
+    return text;
+  }
+
+  /* eslint-disable class-methods-use-this */
```

Comment:
Do we want this in the eslintrc, like it is in studio-frontend?

Reply:
I just cut `eslint-config-edx@4.0.1` so you can add that whenever and then these `/* eslint-disable class-methods-use-this */` statements can be removed.